File: C#/src/Radix/Data/String.cs
using Radix.Math.Pure.Algebra.Operations;
using Radix.Math.Pure.Algebra.Structure;

namespace Radix.Data
{


    public record String(string Value) : Alias<string>(Value), Monoid<String, Addition<String>>
    {

        public Addition<String> Combine => new((s, s2) => s + s2);

        public String Identity => string.Empty;

        public String Empty() => string.Empty;

        public static implicit operator String(string s) => new(s);

        public static implicit operator string(String @string) => @string.Value;

        public String Binary(String x, String y) => x + y;
    }
}

File: TypeScript/src/Radix/String.ts
import { Addition } from "./Math/Pure/Algebra/Operations"
import { Monoid } from "./Math/Pure/Algebra/Structure"

export type String = Monoid<string, Addition<string>>

export function create(s: string) : String {
    return {
        id: s,
        combine: (x: string) => s + x
    }
}

export function toString(s: String) {
    return s.id
}
